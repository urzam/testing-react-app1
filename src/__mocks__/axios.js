const axiosMock = jest.genMockFromModule('axios');

let mockResponse = {
    data: { "shops" : 
        [{
            "location": "test location",
            "address": "test address"
        }]
    }
};

const req = () => {
    return new Promise(function(resolve) {
     axiosMock.delayTimer = setTimeout(function(){
         resolve(mockResponse);
     }, 100);   
    });
}

axiosMock.get.mockImplementation(req);

module.exports = axiosMock;