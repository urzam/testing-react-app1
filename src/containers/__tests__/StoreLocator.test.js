import React from 'react';
import { shallow } from 'enzyme';
import StoreLocator from '../StoreLocator';
import axios from 'axios';
import renderer from 'react-test-renderer';

describe("StoreLocator", () => {
  let mountedStoreLocator;

  beforeEach(()=>{
    mountedStoreLocator = shallow(<StoreLocator />);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<StoreLocator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls axios.get in #componentDidMount', () => {
    return mountedStoreLocator.instance().componentDidMount().then(() => {
      expect(axios.get).toHaveBeenCalled();
    })
  })

  it('calls axios.get in #componentDidMount', () => {
    return mountedStoreLocator.instance().componentDidMount().then(() => {
      expect(axios.get).toHaveBeenCalledWith('http://localhost:3000/data/shops.json');
      const buttons = mountedStoreLocator.find('Button');
      expect(buttons.length).toBe(1);      
    })
  })

  it('updates the state with api data', () => {
    return mountedStoreLocator.instance().componentDidMount().then(() => {
      expect(mountedStoreLocator.state()).toHaveProperty('shops',
        [{
            "location": "test location",
            "address": "test address"
        }]
      );
    })
  });

  it('renders without crashing', () => {
    let mountedStoreLocator = shallow(<StoreLocator />);
  });

  it('renders a header', () => {
    const headers = mountedStoreLocator.find('Header');
    expect(headers.length).toBe(1);
  });

  // it('renders three buttons', () => {
  //   const buttons = mountedStoreLocator.find('Button');
  //   expect(buttons.length).toBe(3);
  // });

  it('renders a map', () => {
    const maps = mountedStoreLocator.find('Map');
    expect(maps.length).toBe(1);
  });  

});

describe('choosemap', () => {
  it('updates this.state.currentMap using the location passed to it', ()=> {
    let mountedStoreLocator = shallow(<StoreLocator />);
    let mockEvent = {target: {value: 'testland'}};
    mountedStoreLocator.instance().chooseMap(mockEvent);
    expect(mountedStoreLocator.instance().state.currentMap).toBe('testland.png');
  });
});