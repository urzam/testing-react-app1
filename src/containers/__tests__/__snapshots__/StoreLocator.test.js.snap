// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocator renders correctly 1`] = `
<div>
  <div
    className="Header"
  >
    <img
      alt="coffe"
      src="images/logo.jpg"
    />
  </div>
  <div />
  <div
    className="MapBox"
  >
    <img
      src="images/none.png"
    />
  </div>
</div>
`;
